### PR TITLE
Fixes in solving steady state problems

### PR DESCRIPTION
**Pull request by paszkow**

### CODE REVIEW

**Comment by dweindl** [comment]
Hi Łukasz, thanks for pushing.
@paulstapor Can you take a look at that?

**Comment by paulstapor** [comment]
Sure, I will!

**Comment by FFroehlich** [comment]
changes in vector.h look reasonable to me

File: include/amici/vector.h [hunk 1]
```diff
@@ -33,7 +33,7 @@ class AmiVector {
       */
     AmiVector(std::vector<realtype> rvec)
         : vec(std::move(rvec)),
-          nvec(N_VMake_Serial(static_cast<long int>(rvec.size()), rvec.data()))
+          nvec(N_VMake_Serial(static_cast<long int>(vec.size()), vec.data()))
```
Comment (dweindl) [inline include/amici/vector.h]: :+1:

**Comment by dweindl** [comment]
@paszkow : Łukasz, can you please merge `fix_amici_tests` (or cherry-pick 7eb6a4b70)  into this pull request? After that, tests should pass and you can merge.

**Comment by paszkow** [comment]
@dweindl Done.

**Pull request merged**